src/pages/smart-contracts/stacks-api-client-tutorial.md

```diff
@@ -0,0 +1,727 @@
+
+```js
+{
+  success && <Transaction txId={txId} />;
+}
+```
+
+You should now be able to see an update in the UI if the transaction was successfully added to the blockchain. In real apps, the progress status could be indicated by colors or other UI elements.
+
+## Show recent activities
+
+Similar to the `TransactionApi`, the `AccountsApi` provides easy access to information about a user or a contract. The api will be used in this section to show recent activities on the to-do list network.
+
+### 1. Step: Create RecentActivities component
+
+Create a new file `RecentActivities.jsx` in folder `src/components` and add the following lines:
+
+```js
+import React, { useCallback, useEffect, useState } from 'react';
+import { Text } from '@blockstack/ui';
+import { AccountsApi } from '@stacks/blockchain-api-client';
+import { CONTRACT_ADDRESS, CONTRACT_NAME } from '../assets/constants';
+
+const accountsApi = new AccountsApi();
+
+export const RecentActivities = () => {
+  const [activities, setActivities] = useState();
+
+  const fetchActivities = useCallback(async () => {
+    // fetch activities
+  }, []);
+
+  useEffect(() => {
+    fetchActivities();
+  }, [fetchActivities]);
+
+  console.log({ activities });
+
+  return activities && activities.length > 0 ? (
+    <Flex
+      display="block"
+      position="absolute"
+      bottom="0"
+      width="100%"
+      justifyContent="space-between"
+      px={4}
+      py={3}
+    >
+      <Box px={3} background="#efefef">
+        <Text fontWeight="500" display="block" mb={0} fontSize={3}>
+          Public To-Do List Registry
+        </Text>
+        <Text fontWeight="500" display="block" mb={0} fontSize={2}>
+          Recent Activities:{' '}
+          {activities.map((activity, key) => {
+            <Text key={key}>{JSON.stringify(activity)}</Text>;
+          })}
+        </Text>
+      </Box>
+    </Flex>
+  ) : null;
+};
+```
+
+Then include it in the app by add the following lines to `Apps.jsx` after the `SignIn` and `TodoList` fragment:
+
+```js
+<RecentActivities />
+```
+
+### 2. Step: Use AccountsApi
+
+Now, add a call to the `getAccountTransactions` method. This method can be used for users and for contracts. It returns transactions that are related to the given account. The default is to return the last 20 transactions.
+
+```js
+const fetchActivities = useCallback(async () => {
+  // fetch activities
+  const response = await accountsApi.getAccountTransactions({
+    principal: `${CONTRACT_ADDRESS}.${CONTRACT_NAME}`,
+  });
+  console.log(response);
+}, []);
+```
+
+-> For users, the principal argument is just user's the stacks address.
+
+### 3. Step: Filter for successful register transactions
+
+The returned transactions can be successful or failed `register` or `update` calls, or even the contract deployment transaction. The recent activities should only contain successfull contract calls. Therefore, add a filter function
```

Comment:
"The returned transactions can be successful or failed `register` or `update` calls, or even the contract deployment transaction."

Quite difficult to read. I'd just say, contract calls could fail (and give a sample reason) and should therefore be excluded.